### PR TITLE
Egrismore

### PR DESCRIPTION
**Pull request by egrismore**
Trying to pull my changes to the 1089 lab

### CODE REVIEW

File: egrismore/Lab4.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    for new in str(i):
+        number = int(new)
+        if number >= largest:
+            return False
+        else:
+            largest = number
+    return True
+
+
+def trick():
```
Comment (ktarplee) [inline egrismore/Lab4.py]: can you separate out the trick from the testing of the trick?

Move the for loop and the decrease() check into the unit test and then just perform the trick in this function.

File: egrismore/Lab4.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        else:
+            largest = number
+    return True
+
+
+def trick(i):
+    """shows the magical 1089"""
+    difference = i - flip(i)
+    difference + flip(difference)
+    print(i)
```
Comment (ktarplee) [inline egrismore/Lab4.py]: return the number and then assert that it equals 1089

Comment (ktarplee) [inline egrismore/Lab4.py]: Emily, are you working on fixing this?

Comment (egrismore) [inline egrismore/Lab4.py]: Yeah, I worked on it a lot yesterday. I expected it to be easy but for
some reason I keep getting an assertion error. I am almost certain I am
doing right but it won't pass the nose tests. It is driving me slightly
crazy.

File: egrismore/Test_Lab4.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+from . import Lab4
+
+
+def test_lab4():
+    for i in range(100, 999):
+        if Lab4.decrease(i):
+            Lab4.trick(i)
```
Comment (ktarplee) [inline egrismore/Test_Lab4.py]: replace this with
assert Lab4.trick(i) == 1089

File: egrismore/Lab4.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+            return False
+        else:
+            largest = number
+    return True
+
+
+def trick(i):
+    """shows the magical 1089"""
+    difference = i - flip(i)
+    difference + flip(difference)
```
Comment (ktarplee) [inline egrismore/Lab4.py]: this line does nothing (it is a noop).

File: egrismore/Lab4.py (hunk 2 above)
Comment (ktarplee) [inline egrismore/Lab4.py]: Put tons of print() lines in your code and figure out what is happening.

Comment (egrismore) [inline egrismore/Lab4.py]: That is exactly what I put for my assert line! What is a noop?

Comment (egrismore) [inline egrismore/Lab4.py]: def trick(i):

```
"""shows the magical 1089"""

difference = i - flip(i)

final = difference + flip(difference)

return final
```

from . import Lab4

def test_Lab4():

```
for i in range(100, 999):

    if Lab4.decrease(i):

        assert Lab4.trick(i) == 1089
```

this is what I have right now

Comment (egrismore) [inline egrismore/Lab4.py]: FAIL: egrismore.Test_Lab4.test_Lab4

---

Traceback (most recent call last):

  File "/usr/local/lib/python3.4/site-packages/nose/case.py", line 198, in
runTest

```
self.test(*self.arg)
```

  File "/Users/emilygrismore/CPSC4500test/egrismore/Test_Lab4.py", line 7,
in test_Lab4

```
assert Lab4.trick(i) == 1089
```

AssertionError

---

Ran 3 tests in 0.929s

This is the error I keep getting

Comment (ktarplee) [inline egrismore/Lab4.py]: commit and push what you have I can try it out.

Comment (egrismore) [inline egrismore/Lab4.py]: Ok. I did it. Thank you!

**Comment by ktarplee** [comment]
I fixed your branch for you.

**Comment by egrismore** [comment]
Thank you! I will take a look to see what you changed

**Pull request merged**